src/types.ts

```diff
@@ -0,0 +1,40 @@
+import { DataSourceJsonData } from '@grafana/data';
+import { DataQuery } from '@grafana/schema';
+
+/**
+ * Query used with Cribl Search
+ */
+export interface CriblQuery extends DataQuery {
+  /**
+   * ID of the Cribl saved search
+   */
+  savedQueryId: string;
+  /**
+   * Max number of results to fetch
+   */
+  maxResults: number;
+}
+
+/**
+ * Options configured for each CriblDataSource instance
+ */
+export interface CriblDataSourceOptions extends DataSourceJsonData {
+  /**
+   * Base URL to the Cribl organization/tenant site (i.e. https://main-elated-nash-dkf9udv.cribl-staging.cloud)
```

Comment:
nit: might be nice to use a non-staging URL as an example. I'd suggesting removing the `-staging` part (no need for this to be a REAL tenant url)